// @aws-sdk/client-apigatewaymanagementapi/dist-cjs/models/errors.js
const { ApiGatewayManagementApiServiceException: __BaseException } = require("./ApiGatewayManagementApiServiceException");
exports.ForbiddenException = class ForbiddenException extends __BaseException {
    name = "ForbiddenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ForbiddenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ForbiddenException.prototype);
    }
};
exports.GoneException = class GoneException extends __BaseException {
    name = "GoneException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "GoneException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GoneException.prototype);
    }
};
exports.LimitExceededException = class LimitExceededException extends __BaseException {
    name = "LimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
};
exports.PayloadTooLargeException = class PayloadTooLargeException extends __BaseException {
    name = "PayloadTooLargeException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PayloadTooLargeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PayloadTooLargeException.prototype);
        this.Message = opts.Message;
    }
};
